jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -256,9 +264,43 @@ private Server(
 		this.pool = checkNotNull(pool);
 		this.outputWriterFactories = ImmutableList.copyOf(firstNonNull(outputWriterFactories, ImmutableList.<OutputWriterFactory>of()));
 		this.outputWriters = ImmutableList.copyOf(firstNonNull(outputWriters, ImmutableList.<OutputWriter>of()));
+		this.breaker = new CircuitBreaker()
+				.withFailureThreshold(3, 10)
+				.withSuccessThreshold(1)
+				.withDelay(1, MINUTES)
+				.withTimeout(250, MILLISECONDS)
+				.onOpen(createLogger("Circuit breaker opened for server {}", this, false))
+				.onClose(createLogger("Circuit breaker closed for server {}", this, true))
+				.onHalfOpen(createLogger("Circuit breaker half opened for server {}", this, true));
+	}
+
+	private CheckedRunnable createLogger(final String message, final Server server, final boolean isSuccess) {
+		return new CheckedRunnable() {
```

Comment:
This anonymous class is the same as in CircuitBreakerOutputWriter.
Instead of true/false flag, using the log level would be clearer, yet SLF4J doesn't allow logging with variable log levels.